core/src/main/java/org/secondbase/core/SecondBase.java

```diff
@@ -56,27 +55,11 @@ private void init(final String[] args, final Flags flags) throws SecondBaseExcep
             module.init();
         }
 
-        if (webConsole != null) {
-            try {
-                webConsole.start();
-            } catch (final IOException e) {
-                throw new SecondBaseException("Could not start webconsole.", e);
+        for (final SecondBaseModule module : configurableModule) {
+            if(module.getShutdownHook().isPresent()) {
+                Runtime.getRuntime().addShutdownHook(module.getShutdownHook().get());
```

Comment:
Can't the module itself add the shutdown hook? Why does it need to be coordinated at all? Consul, for instance, has it's own shutdown hook now.